From 56ae52e381f1460f9e06dd5242b63dab54115fe8 Mon Sep 17 00:00:00 2001
From: Li RongQing <roy.qing.li@gmail.com>
Date: Thu, 24 Sep 2015 09:32:32 +0800
Subject: [PATCH] nfsroot: set the length of parameters passed from the kernel
 command line to 1024

although max nfs path length {NFS_MAXPATHLEN} can be 1024, but nfs_root_parms
, which is used to store the kernel command line parameter, is limited to 256,
and the nfs path will truncated when a user inputs a larger nfs path from kernel
cmdline

make the length of the parameters is adjustable, and default value to 1024

Signed-off-by: Li RongQing <roy.qing.li@gmail.com>
---
 fs/nfs/Kconfig   | 6 ++++++
 fs/nfs/nfsroot.c | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/fs/nfs/Kconfig b/fs/nfs/Kconfig
index 4146a61..607d726 100644
--- a/fs/nfs/Kconfig
+++ b/fs/nfs/Kconfig
@@ -213,3 +213,9 @@ config NFS_DEBUG
 	depends on NFS_FS && SUNRPC_DEBUG
 	select CRC32
 	default y
+
+config ROOT_NFS_CMDLINE_LEN
+	int "the length of parameters passed from the kernel command line (256-1025)"
+	range 256 1025
+	depends on ROOT_NFS
+	default 1024
diff --git a/fs/nfs/nfsroot.c b/fs/nfs/nfsroot.c
index 9bc9f04..5228ace 100644
--- a/fs/nfs/nfsroot.c
+++ b/fs/nfs/nfsroot.c
@@ -90,7 +90,7 @@
 #define NFS_DEF_OPTIONS		"vers=2,udp,rsize=4096,wsize=4096"
 
 /* Parameters passed from the kernel command line */
-static char nfs_root_parms[256] __initdata = "";
+static char nfs_root_parms[CONFIG_ROOT_NFS_CMDLINE_LEN] __initdata = "";
 
 /* Text-based mount options passed to super.c */
 static char nfs_root_options[256] __initdata = NFS_DEF_OPTIONS;
-- 
2.0.2

